extensions/amp-story/1.0/amp-story-reaction-poll-binary.js

```diff
@@ -0,0 +1,172 @@
+import {dev, devAssert} from '../../../src/log';
+import {htmlFor} from '../../../src/static-template';
+import {scopedQuerySelector, scopedQuerySelectorAll} from '../../../src/dom';
+import {toArray} from '../../../src/types';
+
+/** @const {string} */
+const TAG = 'amp-story-reaction-poll-binary';
+
+/** @const @enum {number} */
+export const FontSize = {
+  EMOJI: 28,
+  SINGLE_LINE: 16,
+  DOUBLE_LINE: 14,
+};
+
+/**
+ * Generates the template for the binary poll.
+ *
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildBinaryPollTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`
+    <div class="i-amphtml-story-reaction-poll-binary-container"></div>
+  `;
+};
+
+/**
+ * Generates the template for the binary poll option.
+ *
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildOptionTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`
+    <div class="i-amphtml-story-reaction-option">
+      <span class="i-amphtml-story-reaction-option-title"><span></span></span>
+      <span class="i-amphtml-story-reaction-option-percentage-text">0%</span>
+    </div>
+  `;
+};
+
+export class AmpStoryReactionPollBinary extends AmpStoryReaction {
+  /**
+   * @param {!AmpElement} element
+   */
+  constructor(element) {
+    super(element, ReactionType.POLL, /* bounds */ [2, 2]);
+  }
+
+  /** @override */
+  buildCallback() {
+    super.buildCallback();
+    createShadowRootWithStyle(
+      this.element,
+      dev().assertElement(this.rootEl_),
+      CSS
+    );
+  }
+
+  /** @override */
+  buildComponent() {
+    this.rootEl_ = buildBinaryPollTemplate(this.element);
+    this.attachContent_(this.rootEl_);
+    return this.rootEl_;
+  }
+
+  /**
+   * Gets the options and adds them to the element
+   *
+   * @private
+   * @param {Element} root
+   */
+  attachContent_(root) {
+    this.options_.forEach((option) => {
+      root.appendChild(this.generateOption_(option));
+    });
+    this.adaptFontSize_(root);
+  }
+
+  /**
+   * @private
+   * @param {Element} root
+   */
+  adaptFontSize_(root) {
+    this.mutateElement(() => {
+      let largestFontSize = FontSize.EMOJI;
+      toArray(
+        scopedQuerySelectorAll(
+          dev().assertElement(root),
+          '.i-amphtml-story-reaction-option-title > span'
+        )
+      ).forEach((e) => {
+        if (e.textContent.length <= 3 && largestFontSize >= FontSize.EMOJI) {
+          largestFontSize = FontSize.EMOJI;
+        } else if (
+          e./*OK*/ clientHeight <= 30 &&
+          largestFontSize >= FontSize.SINGLE_LINE
+        ) {
+          largestFontSize = FontSize.SINGLE_LINE;
+        } else if (e./*OK*/ clientHeight > 30) {
+          largestFontSize = FontSize.DOUBLE_LINE;
+        }
+      });
+      root.setAttribute(
+        'style',
+        `--post-select-scale-variable: ${(
+          largestFontSize / FontSize.DOUBLE_LINE
+        ).toFixed(2)} !important;`
+      );
```

Comment:
On pre-select, options have font-size:28px. On post-select, the container for the text will scale down to 50%, so by default setting the scale variable to 1 will scale down the font-size to 14px, which corresponds to the double-line option. For that reason, the largestFontSize allowed gets divided by the DOUBLE_LINE font-size.

TLDR: Sets the scale of the text on post-select with default font-size:14px to achieve the desired font-size according to mocks.
| To achieve this | You need this |
|----------------|---------------|
| font-size: 14px | transform: scale(1) |
| font-size: 16px | transform: scale(1.14) |
| font-size: 28px | transform: scale(2) |

Reply:
I tried to explain the code as much as possible, but the main reason why the scaling is done this way is because the component already scales down the option-title container to 50% to make room for the percentage, so the title is restrained by this container that will scale it down with it.

Then, if no specific scaling is done, the font-size would be reduced also by 50% to a perceived 14px, even though the actual style has the value 28px. Therefore, the scale=2 actually just counteracts the container's scale to keep the font-size at a perceived value of 28px. This ends up keeping the font-size from shrinking when it's an emoji, so that the post-interaction size is bigger. The container still shrinks down since the safe space for the title is smaller, but 2x the scale on the title counteracts that.

SG or do you want to discuss alternatives?

Reply:
I think what Jon meant was that we should not end up with a final scale that is over 1, as it usually appears blurred, so given your explanation we should be good.
However this is a bit complicated, can you document this somewhere? I think the `adaptFontSize_` method deserves a good comment on how things work :))

Reply:
Gave this method a good comment on how it works, lmk if I need to add anything else to the comment since I tried to sum up the behavior as concise as possible.